smg-automotive/components-pkg
# Navigation component

References https://autoricardo.atlassian.net/browse/IN-625

## Motivation and context

Creating header as reusable component. All the header links and configuration are encapsulated within component itself. 

API: 

- user: authenticated user
- language: `de|fr|it|en`
- plattform: `as24|ms24` 
- useAbsoluteUrls: `boolean` - controls whether to use the internal links as absolute URLs or relative paths.
- environment: `preprod/prod` - controls the domain for absolute URLs

Notes: 
- There is no option to choose 'en' as language under language menu, since we don't use it on production
- Drawer columns layout is not according to design atm (waiting for some clarification from design team)

![Screenshot 2023-03-10 at 12 48 41](https://user-images.githubusercontent.com/12614035/224308582-fa82375b-d476-4673-9bff-e3471ddf49df.png)

## Before

N/A

## After

Header component

## How to test

https://navigation-component-components-pkg.branch.autoscout24.dev/?path=/story/patterns-navigation-header--professional


- Check desktop/responsive design -  https://www.figma.com/file/8vy6tgk0DV9KSKyJBjiLSH/12.-AS24---Web---Patterns?node-id=384%3A8619&t=wCQQGtRkh7ji9exJ-0 
- Check navigation item urls (they should should be the same as on https://www.autoscout24.ch/de / https://www.motoscout24.ch/de)
- Check all the languages/translations
- Check navigation items for private/professional user
- Check navigation items depending on plattfrom (as24/ms24)
